components/message.php

```diff
@@ -0,0 +1,504 @@
+
+		for ( $i = 0; $i < $assoc_args['count']; $i++ ) {
+			$this->create( array(), array(
+				'from'      => $this->get_random_user_id(),
+				'to'        => $this->get_random_user_id(),
+				'subject'   => sprintf( 'Message Subject - #%d', $i ),
+				'thread-id' => $assoc_args['thread-id'],
+				'silent'    => true,
+			) );
+
+			$notify->tick();
+		}
+
+		$notify->finish();
+	}
+
+	/**
+	 * Star a message.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --message-id=<message-id>
+	 * : Message ID to star.
+	 *
+	 * --user-id=<user>
+	 * : User that is starring the message. Accepts either a user_login or a numeric ID.
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     $ wp bp message star --message-id=3543 --user-id=user_login
+	 *     Success: Message was successfully starred.
+	 */
+	public function star( $args, $assoc_args ) {
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		$user_id = $user->ID;
+		$msg_id  = (int) $assoc_args['message-id'];
+
+		if ( bp_messages_is_message_starred( $msg_id, $user_id ) ) {
+			WP_CLI::error( 'The message is already starred.' );
+		}
+
+		$star_args = array(
+			'action'     => 'star',
+			'message_id' => $msg_id,
+			'user_id'    => $user_id,
+		);
+
+		if ( bp_messages_star_set_action( $star_args ) ) {
+			WP_CLI::success( 'Message was successfully starred.' );
+		} else {
+			WP_CLI::error( 'Message was not starred.' );
+		}
+	}
+
+	/**
+	 * Unstar a thread.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --thread-id=<thread-id>
+	 * : Thread ID to unstar.
+	 *
+	 * --user-id=<user>
+	 * : User that is unstarring the thread. Accepts either a user_login or a numeric ID.
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     $ wp bp message unstar --thread-id=212 --user-id=another_user_login
+	 *     Success: Message was successfully unstarred.
+	 */
+	public function unstar( $args, $assoc_args ) {
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		$star_args = array(
+			'action'    => 'unstar',
+			'thread_id' => (int) $assoc_args['thread-id'],
+			'user_id'   => $user->ID,
+			'bulk'      => true,
+		);
+
+		if ( bp_messages_star_set_action( $star_args ) ) {
+			WP_CLI::success( 'Message was successfully unstarred.' );
+		} else {
+			WP_CLI::error( 'Message was not unstarred.' );
+		}
+	}
+
+	/**
+	 * Send a notice.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--subject=<subject>]
+	 * : Subject of the notice/message.
+	 * ---
+	 * Default: Random Notice Subject.
+	 * ---
+	 *
+	 * [--content=<content>]
+	 * : Content of the message.
+	 * ---
+	 * Default: Random content.
+	 * ---
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     $ wp bp message send --subject="Important notice" --content="We need to improve"
+	 *     Success: Notice was successfully sent.
+	 *
+	 * @alias send_notice
+	 */
+	public function send( $args, $assoc_args ) {
```

Comment:
`wp bp message send` is a confusing name for this. `send-notice` is probably sufficient, unless you feel like breaking it out into a separate "component" subcommand `wp bp notice`.